frontend/lib/src/dataframes/arrowParseUtils.ts

```diff
@@ -32,49 +32,47 @@ import unzip from "lodash/unzip"
 import { isNullOrUndefined } from "@streamlit/lib/src/util/utils"
 
 import {
-  IndexTypeName,
   isRangeIndex,
-  RangeIndex,
-  Type,
+  PandasColumnType,
+  PandasIndexTypeName,
+  PandasRangeIndex,
 } from "./arrowTypeUtils"
 
 /**
- * A row-major grid of DataFrame index header values.
+ * Index data value.
  */
 type IndexValue = Vector | number[]
 
 /**
- * A row-major grid of DataFrame index header values.
+ * A row-major grid of DataFrame index data values.
  */
-export type Index = IndexValue[]
+export type IndexData = IndexValue[]
 
 /**
  * A row-major grid of DataFrame column header values.
  * NOTE: ArrowJS automatically formats the columns in schema, i.e. we always get strings.
  */
-export type Columns = string[][]
+export type ColumnNames = string[][]
 
 /**
  * A row-major grid of DataFrame data.
  */
 export type Data = Table
 
-/** DataFrame index and data types. */
-export interface Types {
+/** DataFrame's index and data column types. */
+export interface ColumnTypes {
```

Comment:
Since this interface seems to only support Pandas types, should it maybe be called `PandasColumnTypes` similar to the other renamings? And will this be generalized more in a follow-up step?

Reply:
Renamed 👍  

> And will this be generalized more in a follow-up step?

Yep, there is still quite a bit of reliance on pandas metadata to remove to support purely raw arrow data. That will come with follow-up PRs. Goal is that the pandas metadata is only used for additional accuracy, but the main information should come from the Arrow metadata.